src/vs/workbench/contrib/search/browser/search.contribution.ts

```diff
@@ -404,29 +404,82 @@ const FocusSearchListCommand: ICommandAction = {
 };
 MenuRegistry.addCommand(FocusSearchListCommand);
 
-const searchInFolderCommand: ICommandHandler = (accessor, resource?: URI) => {
+const resolveResourcesForSearchIncludes = (resources: URI[], contextService: IWorkspaceContextService): string[] => {
+	resources = distinct(resources, resource => resource.toString());
+
+	const folderPaths: string[] = [];
+	const workspace = contextService.getWorkspace();
+
+	if (resources) {
+		resources.forEach(resource => {
+			let folderPath: string | undefined;
+			if (contextService.getWorkbenchState() === WorkbenchState.FOLDER) {
+				// Show relative path from the root for single-root mode
+				folderPath = relativePath(workspace.folders[0].uri, resource); // always uses forward slashes
+				if (folderPath && folderPath !== '.') {
+					folderPath = './' + folderPath;
+				}
+			} else {
+				const owningFolder = contextService.getWorkspaceFolder(resource);
+				if (owningFolder) {
+					const owningRootName = owningFolder.name;
+
+					// If this root is the only one with its basename, use a relative ./ path. If there is another, use an absolute path
+					const isUniqueFolder = workspace.folders.filter(folder => folder.name === owningRootName).length === 1;
+					if (isUniqueFolder) {
+						const relPath = relativePath(owningFolder.uri, resource); // always uses forward slashes
+						if (relPath === '') {
+							folderPath = `./${owningFolder.name}`;
+						} else {
+							folderPath = `./${owningFolder.name}/${relPath}`;
+						}
+					} else {
+						folderPath = resource.fsPath; // TODO rob: handle on-file URIs
```

Comment:
Do you know what I am talking about in this TODO? This is fine for non-file URIs right? I think I just want the absolute path here if there is another workspace folder with the same name

Reply:
@aeschli gave you that TODO :) https://github.com/microsoft/vscode/commit/a06dbfc3f2a0a69a890583bc77abd36567d5864e I think the issue is using `.fsPath` will be undefined for URI's not backed by the OS's fs.